src/control/lib/telemetry/telemetry.go

```diff
@@ -333,6 +390,38 @@ func Detach(ctx context.Context) {
 	}
 }
 
+func addEphemeralDir(path string, shmSize uint64) error {
+	cPath := C.CString(path)
+	defer C.free(unsafe.Pointer(cPath))
+	if rc := C.add_ephemeral_dir(nil, C.ulong(shmSize), cPath); rc != 0 {
+		return daos.Status(rc)
+	}
+
+	return nil
+}
+
+// SetupClientRoot performs the necessary actions to get the client telemetry
+// segment linked into the agent-managed tree.
+func SetupClientRoot(ctx context.Context, jobid string, pid, shm_key int) error {
+	log := logging.FromContext(ctx)
+
+	if err := addEphemeralDir(jobid, ClientJobMax*C.D_TM_METRIC_SIZE); err != nil {
+		if err != daos.Exists {
+			return errors.Wrapf(err, "failed to add client job path %q", jobid)
+		}
+	}
```

Comment:
Shouldn't need to do this, right? Client is creating their own segment

Reply:
When the agent is managing things, it creates the per-job segment and attaches client process metrics to it under per-pid subdirectories. The clients don't manage or know anything about anything above their per-pid segment.

Reply:
Ahh. I got my lines crossed a bit. So the hierarchy is:

root (created by agent) -> job (created by agent) -> pid (created by libdaos)

Is that correct?

Reply:
Yep, that's it. The unmanaged implementation creates all three of those in the client process, which is obviously not going to work very well when there are multiple users involved.